### PR TITLE
Perf 3: New engine helper

### PR DESCRIPTION
**Pull request by chrisnojima**
New engine helper (engine/saga) for handling back and forth rpc calls. 

1. Only calls that are added to the enabled-calls.json are generated. We tell it which call type we want (channelMap (deprecated), promise, or engineSaga)
1. Better flow typing of calls (caught some stuff already)
1. Fix for WrapAndCall saga helper
1. Removed incomingCallMap in promise style call
1. Use saga channel for avatar loading
1. Dont use parseFolderNameToUsers in chat, just split it
1. When figuring out which convo to auto select use maxBy to go through the list once instead of a filter O(N) + sort O(log2n)

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready for a look

**Comment by chrisnojima** [comment]
@mmaxim you might want to look at this and how it responds to the service call, should be a lot faster to reply

**Comment by chrisnojima** [comment]
@keybase/react-hackers we really should look at getting this in earlier in the release cycle

**Comment by chrisnojima** [comment]
noticed some stuff on ios, deebugging

**Comment by chrisnojima** [comment]
@react-hackers this is ready. It seems fine and no one has said anything for a couple of days of it being on test flight

**Comment by mmaxim** [comment]
Sending animation is broken on TestFlight

**Comment by chrisnojima** [comment]
I think that's an unrelated style issue. I see a few pixels of it

**Comment by chrisnojima** [comment]
yes send animation is broken on master: https://github.com/keybase/client/pull/12299

**Comment by chrisnojima** [comment]
@keybase/react-hackers lets get this in today

File: shared/local-debug.native.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ if (__DEV__) {
   config.printOutstandingTimerListeners = true
   config.printRPC = true
   config.reduxSagaLoggerMasked = false
-  config.userTimings = true
+  config.userTimings = false
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/local-debug.native.js]: measure was constantly throwing caught exceptions so i turned it off, turned back on

**Pull request merged**